### PR TITLE
Ffa updates

### PR DESCRIPTION
**Pull request by jenswi-linaro**

### CODE REVIEW

File: core/arch/arm/kernel/entry_a64.S [hunk 1]
```diff
@@ -56,11 +56,20 @@
 	.endm
 
 FUNC _start , :
+#if defined(CFG_CORE_SEL1_SPMC)
+	/*
+	 * With OP-TEE as SPMC at S-EL1 the SPMD (SPD_spmd) in TF-A passes
+	 * the DTB in x0 and the rest of the registers are unused
+	 */
+	mov	x19, xzr	/* Pagable part not set */
```
Comment (jforissier) [inline core/arch/arm/kernel/entry_a64.S:64]: Does it mean CFG_CORE_SEL1_SPMC cannot be used with pager enabled?

**Review by jforissier — COMMENTED** [review]
One question, in any case:
`Acked-by: Jerome Forissier <jerome@forissier.org>`

File: core/arch/arm/kernel/entry_a64.S (hunk 1 above)
Comment (jenswi-linaro) [reply core/arch/arm/kernel/entry_a64.S:64]: You're right. I've changed my mind, I'll enable pager support too.

**Comment by jenswi-linaro** [comment]
Tags applied. Enabled pager in "core: ffa: FF-A specific boot arguments".

File: core/arch/arm/kernel/entry_a64.S [hunk 2]
```diff
@@ -56,11 +56,20 @@
 	.endm
 
 FUNC _start , :
+#if defined(CFG_CORE_SEL1_SPMC)
+	/*
+	 * With OP-TEE as SPMC at S-EL1 the SPMD (SPD_spmd) in TF-A passes
+	 * the DTB in x0 and the rest of the registers are unused
```
Comment (jforissier) [inline core/arch/arm/kernel/entry_a64.S:62]: `x1` is pageable part

What about the a32 implementation? Shouldn't `r1` play that role too?

Comment (jenswi-linaro) [reply core/arch/arm/kernel/entry_a64.S:62]: You're right. But AArch32 hasn't been tested with FF-A yet, I don't have a working prototype for that. I've just made the corresponding changes.

**Comment by jenswi-linaro** [comment]
Updated arm32 boot args too. Note that this is untested. I tried booting arm32 and it panicked a bit later on something else, but the argument passing seems OK at at least.

File: core/arch/arm/kernel/entry_a64.S (hunk 2 above)
Comment (jforissier) [reply core/arch/arm/kernel/entry_a64.S:62]: Sounds reasonable.
Back to the a64 code, this comment still needs to be updated "DTB in x0 and the rest of the registers are unused", wrong, x1 is used.

Comment (jenswi-linaro) [reply core/arch/arm/kernel/entry_a64.S]: Sorry, I'll fix.

**Comment by jenswi-linaro** [comment]
Fixed up register usage for both arm64 and arm32.

**Pull request merged**